include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -223,172 +358,428 @@ namespace sparrow
+        if (!is_created_with_sparrow())
         {
-            case buffer_type::VALIDITY:
-                return static_cast<std::size_t>(std::ceil(static_cast<double>(length) / bit_per_byte));
-            case buffer_type::DATA:
-                return primitive_bytes_count(data_type, length);
-            case buffer_type::OFFSETS_32BIT:
-            case buffer_type::SIZES_32BIT:
-                return get_offset_size(data_type, length) * sizeof(std::int32_t);
-            case buffer_type::OFFSETS_64BIT:
-            case buffer_type::SIZES_64BIT:
-                return get_offset_size(data_type, length) * sizeof(std::int64_t);
-            case buffer_type::VIEWS:
-                // TODO: Implement
-                SPARROW_ASSERT(false, "Not implemented");
-                return 0;
-            case buffer_type::TYPE_IDS:
-                return static_cast<std::size_t>(length);
+            throw arrow_proxy_exception("Cannot set n_buffers on non-sparrow created ArrowArray");
         }
-        mpl::unreachable();
+        m_array_ptr->n_buffers = static_cast<int64_t>(n_buffers);
+        arrow_array_private_data* private_data = get_array_private_data();
+        private_data->resize_buffers(n_buffers);
+        m_array_ptr->buffers = private_data->buffers_ptrs<void>();
+        m_array_ptr->n_buffers = static_cast<int64_t>(n_buffers);
     }
 
-    [[nodiscard]] inline std::vector<sparrow::buffer_view<uint8_t>>& arrow_proxy::buffers()
+    [[nodiscard]] inline size_t arrow_proxy::n_children() const
     {
-        return m_buffers;
+        return static_cast<size_t>(m_array_ptr->n_children);
+    }
+
+    template <std::ranges::input_range R>
+        requires std::same_as<std::ranges::range_value_t<R>, arrow_array_and_schema_pointers>
+    inline void arrow_proxy::add_children(const R& arrow_array_and_schema_pointers)
+    {
+        if (!is_created_with_sparrow())
+        {
+            throw arrow_proxy_exception("Cannot set n_buffers on non-sparrow created ArrowArray");
+        }
+
+        const size_t add_children_count = std::ranges::size(arrow_array_and_schema_pointers);
+        const size_t original_children_count = n_children();
+        const size_t new_children_count = original_children_count + add_children_count;
+
+        resize_children(new_children_count);
+        for (size_t i = 0; i < add_children_count; ++i)
+        {
+            set_child(
+                i + original_children_count,
+                arrow_array_and_schema_pointers[i].array,
+                arrow_array_and_schema_pointers[i].schema
+            );
+        }
+    }
+
+    inline void arrow_proxy::pop_children(size_t n)
+    {
+        if (!is_created_with_sparrow())
+        {
+            throw arrow_proxy_exception("Cannot set n_buffers on non-sparrow created ArrowArray");
+        }
+
+        if (n > n_children())
+        {
+            throw arrow_proxy_exception("Cannot pop more children than the current number of children");
```

Comment:
This function is not documented as such. Also, an alternative behavior would be to pop as much as possible (`std::min(n, n_children())`) which is documented would be reasonable too. Anyway, it would be useful to document the prerequisites and behavior for this function.

Reply:
Documentation added